### ISSUE TITLE
Assertion failed in libtorrent::utp_socket_manager::send_packet

### ISSUE THREAD

**Issue by vktr** [arvidn/libtorrent#591]
version: 1.1.0.0-7f05551

File: `src\utp_socket_manager.cpp`
Line: 167
Function: `libtorrent::utp_socket_manager::send_packet`
Expression: !tmp
tmp.message(): An invalid argument was supplied

Stack

```
0: print_backtrace
1: assert_fail
2: libtorrent::utp_socket_manager::send_packet
3: libtorrent::utp_socket_impl::send_pkt
4: libtorrent::utp_stream::issue_write
5: libtorrent::utp_stream::async_write_some<std::vector<boost::asio::const_buffer,std::allocator<boost::asio::const_buffer> >,libtorrent::aux::allocating_handler<boost::_bi::bind_t<void,boost::_mfi::mf2<void,libtorrent::peer_connection,boost::system::error_code const & __ptr64,unsigned __int64>,boost::_bi::list3<boost::_bi::value<boost::shared_ptr<libtorrent::peer_connection> >,boost::arg<1>,boost::arg<2> > >,336> >
6: libtorrent::socket_type::async_write_some<std::vector<boost::asio::const_buffer,std::allocator<boost::asio::const_buffer> >,libtorrent::aux::allocating_handler<boost::_bi::bind_t<void,boost::_mfi::mf2<void,libtorrent::peer_connection,boost::system::error_code const & __ptr64,unsigned __int64>,boost::_bi::list3<boost::_bi::value<boost::shared_ptr<libtorrent::peer_connection> >,boost::arg<1>,boost::arg<2> > >,336> >
7: libtorrent::peer_connection::setup_send
8: libtorrent::peer_connection::uncork_socket
9: libtorrent::cork::~cork
10: libtorrent::bt_peer_connection::on_connected
11: libtorrent::peer_connection::on_connection_complete
12: boost::_mfi::mf1<void,libtorrent::peer_connection,boost::system::error_codeconst & __ptr64>::call<boost::shared_ptr<libtorrent::peer_connection>,boost::system::error_code const >
13: boost::_mfi::mf1<void,libtorrent::peer_connection,boost::system::error_codeconst & __ptr64>::operator()<boost::shared_ptr<libtorrent::peer_connection> >
14: boost::_bi::list2<boost::_bi::value<boost::shared_ptr<libtorrent::peer_connection> >,boost::arg<1> >::operator()<boost::_mfi::mf1<void,libtorrent::peer_connection,boost::system::error_code const & __ptr64>,boost::_bi::rrlist1<boost::system::error_code const & __ptr64> >
15: boost::_bi::bind_t<void,boost::_mfi::mf1<void,libtorrent::peer_connection,boost::system::error_code const & __ptr64>,boost::_bi::list2<boost::_bi::value<boost::shared_ptr<libtorrent::peer_connection> >,boost::arg<1> > >::operator()<boost::system::error_code const & __ptr64>
16: boost::detail::function::void_function_obj_invoker1<boost::_bi::bind_t<void,boost::_mfi::mf1<void,libtorrent::peer_connection,boost::system::error_code const & __ptr64>,boost::_bi::list2<boost::_bi::value<boost::shared_ptr<libtorrent::peer_connection> >,boost::arg<1> > >,void,boost::system::error_code const & __ptr64>::invoke
17: boost::function1<void,boost::system::error_code const & __ptr64>::operator()
18: boost::_bi::list1<boost::_bi::value<boost::system::error_code> >::operator()<boost::function1<void,boost::system::error_code const & __ptr64>,boost::_bi::list0>
19: boost::_bi::bind_t<void,boost::function1<void,boost::system::error_code const & __ptr64>,boost::_bi::list1<boost::_bi::value<boost::system::error_code> > >::operator()
20: boost::asio::asio_handler_invoke<boost::_bi::bind_t<void,boost::function1<void,boost::system::error_code const & __ptr64>,boost::_bi::list1<boost::_bi::value<boost::system::error_code> > > >
21: boost_asio_handler_invoke_helpers::invoke<boost::_bi::bind_t<void,boost::function1<void,boost::system::error_code const & __ptr64>,boost::_bi::list1<boost::_bi::value<boost::system::error_code> > >,boost::_bi::bind_t<void,boost::function1<void,boost::system::error_code const & __ptr64>,boost::_bi::list1<boost::_bi::value<boost::system::error_code> > > >
22: boost::asio::detail::completion_handler<boost::_bi::bind_t<void,boost::function1<void,boost::system::error_code const & __ptr64>,boost::_bi::list1<boost::_bi::value<boost::system::error_code> > > >::do_complete
23: boost::asio::detail::win_iocp_operation::complete
24: boost::asio::detail::win_iocp_io_service::do_one
25: boost::asio::detail::win_iocp_io_service::run
26: boost::asio::io_service::run
27: boost::_mfi::mf0<unsigned __int64,boost::asio::io_service>::operator()
28: boost::_bi::list1<boost::_bi::value<boost::asio::io_service * __ptr64> >::operator()<unsigned __int64,boost::_mfi::mf0<unsigned __int64,boost::asio::io_service>,boost::_bi::list0>
29: boost::_bi::bind_t<unsigned __int64,boost::_mfi::mf0<unsigned __int64,boost::asio::io_service>,boost::_bi::list1<boost::_bi::value<boost::asio::io_service * __ptr64> > >::operator()
30: boost::asio::detail::win_thread::func<boost::_bi::bind_t<unsigned __int64,boost::_mfi::mf0<unsigned __int64,boost::asio::io_service>,boost::_bi::list1<boost::_bi::value<boost::asio::io_service * __ptr64> > > >::run
31: boost::asio::detail::win_thread_function
32: invoke_thread_procedure
33: thread_start<unsigned int (__cdecl*)(void * __ptr64)>
34: BaseThreadInitThunk
35: RtlUserThreadStart
```

**Comment by ssiloti** [comment]
This was fixed by 4949721389fdb00b29cec1d959aaf21bc863abdc

**Comment by vktr** [comment]
Thanks, pulling the latest sources fixed it 😄
